core/base/inc/ROOT/TExecutorBaseImpl.hxx

```diff
@@ -9,21 +9,21 @@
  * For the list of contributors see $ROOTSYS/README/CREDITS.             *
  *************************************************************************/
 
-#ifndef ROOT_TExecutor
-#define ROOT_TExecutor
+#ifndef ROOT_TExecutorBaseImpl
+#define ROOT_TExecutorBaseImpl
 
 #include "ROOT/TSeq.hxx"
 #include "TList.h"
 #include <vector>
 
 //////////////////////////////////////////////////////////////////////////
 ///
-/// \class ROOT::TExecutor
+/// \class ROOT::TExecutorBaseImpl
```

Comment:
why `BaseImpl`? I'd suggest `TExecutorCRTP` to clarify that the only role of this class is to decorate derived classes with some common extra behavior.